pemFioi/quickpi/blocklyQuickPi_lib.js

```diff
@@ -1745,8 +1749,121 @@ var getContext = function (display, infos, curLevel) {
             conceptList.push(concept);
         }
         return conceptList;
-    }
+    };
+
+    /**
+     * This method allow us to find all the functions according to the available sensors
+     */
+    context.findAllSensorsFunctions = function() {
+        // alwaysIncluded are sensors that are always included and hidden to the user.
+        var alwaysIncluded = ["quickpi_wait"];
+        var quickPiConceptList = context.quickpi.conceptList;
+        var ret = [];
+        var sensorTypes = (function() {
+            var sensors = infos.quickPiSensors;
+            var retSensorType = [];
+            for (var iSensor = 0; iSensor < sensors.length; iSensor++)
+                retSensorType.push("quickpi_" + sensors[iSensor].type);
+            for (var i = 0; i < alwaysIncluded.length; i++)
+                retSensorType.push(alwaysIncluded[i]);
+
+            // cleaning output so it correspond to the concept output
+            if (arrayContains(retSensorType, "quickpi_cloudstore"))
```

Comment:
You can simplify all these lines by having an object with all the sensorType -> concept name pairs, and iterating over it.

Reply:
You are right, didn't thought about it before, thanks, that is what I have done for next commit.